aten/src/ATen/core/TransformationHelper.h

```diff
@@ -74,4 +76,49 @@ C10_HOST_DEVICE inline dist_acctype<T> uniform_real(V val, T from, T to) {
   return (x * (to - from) + from);
 }
 
+/**
+ * Transforms normally distributed `val` with mean 0.0 and standard deviation 1.0 to
+ * normally distributed with `mean` and standard deviation `std`.
+ */
+template <typename T>
+C10_HOST_DEVICE inline T normal(T val, T mean, T std) {
+  return val * std + mean;
+}
+
+/**
+ * Transforms uniformly distributed `val` between 0.0 and 1.0 to
+ * Cauchy distribution with location parameter `median` and scale parameter `sigma`.
+ */
+template <typename T>
+C10_HOST_DEVICE inline T cauchy(T val, T median, T sigma) {
+  return median + sigma * at::tan(static_cast<T>(M_PI) * (val - static_cast<T>(0.5)));
+}
+
+/**
+ * Transforms uniformly distributed `val` between 0.0 and 1.0 to
+ * exponentialy distributed with `lambda` parameter of the distribution.
+ */
+template <typename T>
+C10_HOST_DEVICE __ubsan_ignore_float_divide_by_zero__ inline T exponential(T val, T lambda) {
+  return static_cast<T>(-1.0) / lambda * at::log(static_cast<T>(1.0) - val);
```

Comment:
It is correct from math perspective but may affect exponential distribution because of different float precision near 0 and 1, so I'd prefer to leave it as is